numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -476,18 +472,29 @@ UNICODE_setitem(PyObject *op, void *ov, void *vap)
     else if ((temp=PyObject_Str(op)) == NULL) {
         return -1;
     }
-    ptr = PyUnicode_AS_UNICODE(temp);
-    if ((ptr == NULL) || (PyErr_Occurred())) {
+
+    /* needs to be in canonical form for PyUnicode_GET_LENGTH below */
+    if (PyUnicode_READY(temp) < 0) {
```

Comment:
I was curious if it is really necessary, but the cpython code is not that quickly figured out :). Nvm, lets just have it there in any case.

Reply:
I could switch to the non-macro form of `GetLength` here, this is a remnant of a broken strategy.

Reply:
Made that switch

Reply:
Not sure what is better, but whatever, this is good :).